halogen/schema.py

```diff
@@ -421,6 +439,8 @@ def serialize(cls, value, **kwargs):
             except (AttributeError, KeyError):
                 if attr.required:
```

Comment:
this should now always raise, not only if it's required.

Reply:
If its not required we do not need to raise anything if the attribute is not given.